plugins/woocommerce/includes/class-wc-post-types.php

```diff
@@ -123,6 +123,7 @@ public static function register_taxonomies() {
 						'with_front'   => false,
 						'hierarchical' => true,
 					),
+					'meta_box_cb'           => 'WC_Meta_Box_Product_Categories::output',
```

Comment:
@louwie17 I'm seeing really odd behavior...

With this set as such, saving does not work properly, with either the new typeahead field, or the existing field. It seems to be using the category `name` as the `term_id` when saving?!

If I change this to...

```
'meta_box_cb => 'post_categories_meta_box'
```

Then things work properly (it always uses the old field, of course).

I tried debugging this a bit, but I can't figure out what is happening!

Reply:
@louwie17 This issue was really haunting me... I couldn't get it out of my brain and needed to understand what was going on... I dug more into the underlying WP code and determined the issue...

We need to set 

```
'meta_box_sanitize_cb' => 'taxonomy_meta_box_sanitize_cb_checkboxes',
```

Because, without doing this, `meta_box_sanitize_cb` defaults to "an appropriate one based on the value of `$meta_box_cb`" ([doc](https://developer.wordpress.org/reference/functions/register_taxonomy/)), which ends up being `taxonomy_meta_box_sanitize_cb_input` for any `$meta_box_cb` that isn't `post_categories_meta_box` ([src](https://github.com/WordPress/wordpress-develop/blob/e0005a5788630717740ccb7150ae39d0feeae789/src/wp-includes/class-wp-taxonomy.php#L452)).

I tested this change with both the existing category field and the new typeahead field and things appear to function properly now!

Please make this change and test it youself to verify that the save works as intended, and then let me know and I'll give this a re-re-re-review. 😄